### PR TITLE
Add OAuth2 authentication

### PR DESCRIPTION
**Pull request by 9a24f0**
Resolve GH-68
Also returns Token for later API Call.

### CODE REVIEW

**Comment by 9a24f0** [comment]
Eh, actually `Session.kt` is not needed anymore

**Comment by 9a24f0** [comment]
Seems like something not working for sign out, but merging this should handle the token problem for other activities so...

**Review by Huy-Ngo — APPROVED** [review]
The toast seems to indicate that this works.
It's ugly but it worked

File: app/src/main/java/vn/edu/usth/minigh/AuthActivity.kt [hunk 1]
```diff
@@ -1,21 +1,176 @@
...
+            val githubId = jsonObject.getInt("id")
+            Log.i("GitHub Id: ", githubId.toString())
+            id = githubId.toString()
+
+            // GitHub Display Name
+            val githubDisplayName = jsonObject.getString("login")
+            Log.i("GitHub Display Name: ", githubDisplayName)
+            displayName = githubDisplayName
+        }
     }
```
Comment (9a24f0) [inline app/src/main/java/vn/edu/usth/minigh/AuthActivity.kt]: oi @Huy-Ngo ,@PhuongPhg and @s0ys4uc3, you can take this as example for getting stuff

**Comment by McSinyx** [comment]
Isn't most stuff through REST API can be retrieved via Retrofit?
https://square.github.io/retrofit/2.x/retrofit/retrofit2/http/Header.html

**Comment by Huy-Ngo** [comment]
> Isn't most stuff through REST API can be retrieved via Retrofit?
@McSinyx 

Something like this?

```kotlin
     @GET("/user")
     suspend fun current_user(@Header("Authorization" token: String)): User

```

**Comment by McSinyx** [comment]
Yes, but don't forget that the token is prefixed by `token `.

**Pull request merged**